mmengine/runner/runner.py

```diff
@@ -349,19 +346,34 @@ def __init__(
         self._randomness_cfg = randomness
         self.set_randomness(**randomness)
 
-        if experiment_name is not None:
-            self._experiment_name = f'{experiment_name}_{self._timestamp}'
-        elif self.cfg.filename is not None:
+        if experiment_name is None and self.cfg.filename is not None:
             filename_no_ext = osp.splitext(osp.basename(self.cfg.filename))[0]
-            self._experiment_name = f'{filename_no_ext}_{self._timestamp}'
-        else:
-            self._experiment_name = self.timestamp
-        self._log_dir = osp.join(self.work_dir, self.timestamp)
+            experiment_name = f'{filename_no_ext}'
+
+        self._experiment_name = experiment_name
+        self._experiment_id = self._timestamp
+
+        if work_dir is None:
+            if 'work_dir' in self.cfg and self.cfg.work_dir is not None:
+                work_dir = self.cfg.work_dir
+            elif self._experiment_name is not None:
+                work_dir = osp.join('./work_dirs', self._experiment_name)
+            else:
+                raise ValueError(
+                    'work_dir, cfg.work_dir, experiment_name and cfg.filename'
+                    'can not be all None.')
```

Comment:
Should we make the f`work_dir/{self.timestamp}` as the default `work_dir`?

Reply:
Do you mean that we change
```python
        self._work_dir = osp.abspath(work_dir)
        if 'auto_distinguish_experiment' in self.cfg \
                and self.cfg.auto_distinguish_experiment:
            self._work_dir = osp.join(self._work_dir, self._experiment_id)
```
to
```python
        self._work_dir = osp.abspath(work_dir, , self._experiment_id)
```
and remove the auto_distinguish_experiment flag?

I can accept it, but what do you and your colleagues think about it?

Reply:
It souds great to me.